xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/job/DeleteJob.java

```diff
@@ -111,16 +126,24 @@ public void visit(DocumentReference documentReference)
 
     private void maybeDelete(DocumentReference documentReference)
     {
-        EntitySelection entitySelection = concernedEntities.get(documentReference);
+        Boolean skipRecycleBin = this.getRequest().getProperty(SKIP_RECYCLE_BIN_PROPERTY);
+        boolean toRecycleBin = !this.configuration.canSkipRecycleBin()
```

Comment:
it's really error prone here that you mix "skipRecycleBin" and "toRecycleBin" IMO.

Reply:
Replace with `skipRecycleBin` here https://github.com/xwiki/xwiki-platform/pull/1369/files#diff-57f9752a12af2ec241f5cb8a244f427fR130